src/Cms/Roles.php

```diff
@@ -97,15 +97,21 @@ public static function factory(array $roles, array $inject = [])
      */
     public static function load(string $root = null, array $inject = [])
     {
+        $kirby = App::instance();
         $roles = new static();
 
         // load roles from plugins
-        foreach (App::instance()->extensions('blueprints') as $blueprintName => $blueprint) {
+        foreach ($kirby->extensions('blueprints') as $blueprintName => $blueprint) {
             if (substr($blueprintName, 0, 6) !== 'users/') {
                 continue;
             }
 
             if (is_array($blueprint) === true) {
+                $role = Role::factory($blueprint, $inject);
+            } elseif (is_callable($blueprint) === true) {
+                $blueprint = $blueprint($kirby);
+                $blueprint['name'] ??= $blueprintName;
```

Comment:
I'd make this a separate if statement before the others. Keeps code a bit more DRY and would allow also returning a string form the callback

Reply:
I did exactly as you said at the beginning. But that would be inconsistent as it would be different from page blueprint loads. That's why I did it this way. But if you want, I can develop the callback string to return on page blueprint loads as well.

Reply:
As far as I can tell, the only difference would be that the current behavior only allows arrays to be returned from the callback whereas a separate `if` statement before would also allow the `else` case with a string. Would anything speak against supporting that?

Reply:
I have no contrary opinion. Yes, it is very easy to implement. I think it would be useful to return a file path, not just an array. If everyone agrees, I can make this development for both parts.

Reply:
I agree, if we do it here, we should also support it for page blueprints.